src/viam/ur/module/ur_arm.cpp

```diff
@@ -71,6 +76,68 @@ pose ur_vector_to_pose(urcl::vector6d_t vec) {
     return {position, orientation, theta};
 }
 
+urcl::vector6d_t pose_to_ur_vector(const pose& p) {
+    if (!std::isfinite(p.theta)) {
+        throw std::invalid_argument("pose_to_ur_vector: theta is infinite or NaN");
+    }
+
+    urcl::vector6d_t v{};
+
+    // convert millimeters to meters
+    v[0] = p.coordinates.x / 1000.0;
+    v[1] = p.coordinates.y / 1000.0;
+    v[2] = p.coordinates.z / 1000.0;
+
+    const double theta_rad = degrees_to_radians(p.theta);
+
+    // If angle ~ 0, UR expects the axis-angle "vector" to be zero.
+    constexpr double k_angle_epsilon = 1e-12;
+    if (std::abs(theta_rad) < k_angle_epsilon) {
+        v[3] = v[4] = v[5] = 0.0;
+        return v;
+    }
+
+    // convert viam's orientation to axis angles which is what universal robots use
+    // viam orientation vector -> quaternion -> axis angle -> universal robots orientation vector
+    // create an orientation vector to use
+    auto ov = std::unique_ptr<void, decltype(&free_orientation_vector_memory)>(
+        new_orientation_vector(p.orientation.o_x, p.orientation.o_y, p.orientation.o_z, theta_rad), &free_orientation_vector_memory);
+    if (!ov) {
+        throw std::runtime_error("pose_to_ur_vector: failed to create orientation vector");
+    }
+
+    // create quaternion from the orientation vector
+    auto q =
+        std::unique_ptr<void, decltype(&free_quaternion_memory)>(quaternion_from_orientation_vector(ov.get()), &free_quaternion_memory);
+    if (!q) {
+        throw std::runtime_error("pose_to_ur_vector: failed to create quaternion");
+    }
+
+    // convert the quaternion to axis angles
+    auto aa = std::unique_ptr<double[], decltype(&free_axis_angles_memory)>(axis_angle_from_quaternion(q.get()), &free_axis_angles_memory);
+    if (!aa) {
+        throw std::runtime_error("pose_to_ur_vector: failed to compute axis-angle");
+    }
+
+    const double ax = aa[0];
+    const double ay = aa[1];
+    const double az = aa[2];
+    const double th = aa[3];
+
+    // robustness to tiny axes/angles
+    const double n = std::hypot(ax, ay, az);
+    if (n < k_angle_epsilon || std::abs(th) < k_angle_epsilon) {
```

Comment:
`n` is a length, but `k_angle_epsilon` is in radians. Is the comparison meaningful?